langchain/src/prompts/chat.ts

```diff
@@ -559,17 +559,58 @@ export class ChatPromptTemplate<
     return "chat";
   }
 
+  private async _parseImagePrompts(
+    message: BaseMessage,
+    inputValues: InputValues<
+      PartialVariableName | Extract<keyof RunInput, string>
+    >
+  ): Promise<BaseMessage> {
+    if (typeof message.content === "string") {
+      return message;
+    }
+    const formattedMessageContent = await Promise.all(
+      message.content.map(async (item) => {
+        if (
+          item.type !== "image_url" ||
+          typeof item.image_url === "string" ||
+          !item.image_url?.url
+        ) {
+          return item;
+        }
+        const imageUrl = item.image_url.url;
+        const promptTemplatePlaceholder = PromptTemplate.fromTemplate(imageUrl);
+        const formattedUrl = await promptTemplatePlaceholder.format(
+          inputValues
+        );
+        // eslint-disable-next-line no-param-reassign
+        item.image_url.url = formattedUrl;
+        return item;
+      })
+    );
+    // eslint-disable-next-line no-param-reassign
+    message.content = formattedMessageContent;
+    return message;
+  }
+
   async formatMessages(
     values: TypedPromptInputValues<RunInput>
   ): Promise<BaseMessage[]> {
     const allValues = await this.mergePartialAndUserVariables(values);
-
     let resultMessages: BaseMessage[] = [];
 
     for (const promptMessage of this.promptMessages) {
       // eslint-disable-next-line no-instanceof/no-instanceof
       if (promptMessage instanceof BaseMessage) {
-        resultMessages.push(promptMessage);
+        if (
+          typeof promptMessage.content !== "string" &&
+          promptMessage.content.some((content) => content.type === "image_url")
+        ) {
```

Comment:
I don't think we need a branch here? Looks like we can handle it all in one method and just return if it's a string?

Reply:
Yes good call